### PR TITLE
Batch 7: Kokomo

### PR DESCRIPTION
**Pull request by ericsorenson**
- 8809 [BUG: Order Search: Docket number and Case title/Petitioner Name field search validation fires incorrectly after manually clearing the field](https://github.com/flexion/ef-cms/issues/8809) 
- 8804 [Search: User Feedback](https://github.com/flexion/ef-cms/issues/8804) 
- 8725 [Search: Usability Improvement Package ustaxcourt/ef-cms#6491](https://github.com/flexion/ef-cms/issues/8725) 
- 8682 [Order Search: Non-Keyword/Phrase Searches](https://github.com/flexion/ef-cms/issues/8682) 
- 8640 [Court: Remove Notice of Change of Address/Phone/Both from eFiled document list](https://github.com/flexion/ef-cms/issues/8640) 
- 8552 [Order Search: Add more descriptive instructions](https://github.com/flexion/ef-cms/issues/8552) 
- 8545 [Search - Logging Order Search User Data v1](https://github.com/flexion/ef-cms/issues/8545) 
- 8544 [Order Search: Turn on Order Search internally in the production environment](https://github.com/flexion/ef-cms/issues/8544) 
- 8535 [SPIKE: Assist with documentation of decision to use DynamoDB](https://github.com/flexion/ef-cms/issues/8535) 

A note for story ustaxcourt/ef-cms#7948 - there is a script `scripts/setup-order-search-flag.sh` that will need to be run for each environment to set the order search flag to true in the deploy table. An environment-specific deploy will also need to be run for each environment.

### CODE REVIEW

File: shared/src/business/entities/caseAssociation/AddPrivatePractitionerFactory.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ const { SERVICE_INDICATOR_TYPES } = require('../EntityConstants');
  * @constructor
  */
 function AddPrivatePractitionerFactory(rawProps) {
+  /**
+   *
```
Comment (mmarcotte) [inline shared/src/business/entities/caseAssociation/AddPrivatePractitionerFactory.js]: Could we fill in this void?

File: shared/src/business/entities/caseAssociation/EditPetitionerCounselFactory.js [hunk 2]
```diff
@@ -17,6 +17,9 @@ const {
  * @constructor
  */
 function EditPetitionerCounselFactory(rawProps) {
+  /**
+   *
```
Comment (mmarcotte) [inline shared/src/business/entities/caseAssociation/EditPetitionerCounselFactory.js]: Could we fill in this void?

File: shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js [hunk 3]
```diff
@@ -126,6 +126,9 @@ const VALIDATION_ERROR_MESSAGES = {
  * @constructor
  */
 function ExternalDocumentInformationFactory(documentMetadata) {
+  /**
+   *
```
Comment (mmarcotte) [inline shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js]: Could we fill in this void?

File: shared/src/business/entities/externalDocument/SecondaryDocumentInformationFactory.js [hunk 4]
```diff
@@ -20,6 +20,9 @@ function SecondaryDocumentInformationFactory(
   documentMetadata,
   VALIDATION_ERROR_MESSAGES,
 ) {
+  /**
+   *
```
Comment (mmarcotte) [inline shared/src/business/entities/externalDocument/SecondaryDocumentInformationFactory.js]: Could we fill in this void?

File: shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js [hunk 5]
```diff
@@ -52,13 +52,7 @@ exports.deleteCounselFromCaseInteractor = async (
...
   }
 
-  caseEntity.petitioners.forEach(petitioner => {
-    if (
-      !caseEntity.isUserIdRepresentedByPrivatePractitioner(petitioner.contactId)
-    ) {
-      petitioner.serviceIndicator = null;
-    }
-  });
+  caseEntity = setupServiceIndicatorForUnrepresentedPetitioners(caseEntity);
```
Comment (mmarcotte) [inline shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js:55]: refactoring backlog consideration, wouldn't it make sense for this function to be a class method?

File: shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js [hunk 6]
```diff
@@ -317,4 +301,86 @@ describe('fileExternalDocumentForConsolidatedInteractor', () => {
...
+        .workItem,
+    ).toMatchObject({
+      completedMessage: 'completed',
+    });
+  });
+
+  it('should not set as served when is NOT isAutoServed', async () => {
+    applicationContext.getPersistenceGateway().getUserById.mockReturnValue({
+      name: 'Guy Fieri',
+      role: ROLES.admin,
```
Comment (mmarcotte) [inline shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js]: should we be using `ROLES.admin` here?

File: shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js [hunk 7]
```diff
@@ -124,6 +124,32 @@ describe('deleteTrialSessionInteractor', () => {
     expect(applicationContext.getPersistenceGateway().updateCase).toBeCalled();
   });
 
+  it('does not delete the trial session working copy and if there is no judge on the trial session', async () => {
```
Comment (mmarcotte) [inline shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js]: nit: typo with. I think this should be:

>   it('does not delete the trial session working copy if there is no judge on the trial session', async () => {

File: web-client/src/presenter/computeds/AdvancedSearch/advancedSearchHelper.js [hunk 8]
```diff
@@ -22,16 +22,29 @@ export const formatSearchResultRecord = (result, { applicationContext }) => {
...
   const currentPage = get(state.advancedSearchForm.currentPage);
+
+  const showDateRangePicker =
+    dateRangeType === DATE_RANGE_SEARCH_OPTIONS.CUSTOM_DATES;
+  const showFeedbackButton = applicationContext
+    .getUtilities()
+    .isInternalUser(user.role);
+
   const result = {
+    feedBackUrl: 'https://forms.office.com/r/J1AHm7d3BE',
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/AdvancedSearch/advancedSearchHelper.js:45]: feels a little goofy that this is embedded within the client and not obscured by an authenticated api call. basically anybody out there can find this in our source and submit feedback. i guess we'll just address that if it becomes a problem.

File: web-client/src/presenter/computeds/partiesInformationHelper.js [hunk 9]
```diff
@@ -21,8 +21,32 @@ export const formatCounsel = ({ counsel, screenMetadata }) => {
...
+    canEditPetitioner = petitioner.contactId === user.userId;
+  } else if (user.role === USER_ROLES.privatePractitioner) {
+    canEditPetitioner = userAssociatedWithCase;
+  } else if (permissions.EDIT_PETITIONER_INFO) {
+    canEditPetitioner = true;
+  }
+
+  canEditPetitioner = petitionIsServed && canEditPetitioner;
+
+  return canEditPetitioner;
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/partiesInformationHelper.js]: Feels cleaner to just have

```
if (!petitionIsServed) { 
  return false;
}

if (user.role === USER_ROLES.petitioner) {
  return true;
} else if (user.role === USER_ROLES.privatePractitioner) {
  return userAssociatedWithCase;
} else if (permissions.EDIT_PETITIONER_INFO) {  
  return true;
}

return false;
```

File: shared/src/business/entities/caseAssociation/AddPrivatePractitionerFactory.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/entities/caseAssociation/AddPrivatePractitionerFactory.js]: Updated!

File: shared/src/business/entities/caseAssociation/EditPetitionerCounselFactory.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/business/entities/caseAssociation/EditPetitionerCounselFactory.js]: Updated!

File: shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js (hunk 3 above)
Comment (ericsorenson) [reply shared/src/business/entities/externalDocument/ExternalDocumentInformationFactory.js]: Updated!

File: shared/src/business/entities/externalDocument/SecondaryDocumentInformationFactory.js (hunk 4 above)
Comment (ericsorenson) [reply shared/src/business/entities/externalDocument/SecondaryDocumentInformationFactory.js]: Updated!

File: shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js (hunk 6 above)
Comment (ericsorenson) [reply shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js]: Updated!

File: shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js (hunk 7 above)
Comment (ericsorenson) [reply shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js]: Updated!

File: shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js (hunk 5 above)
Comment (ericsorenson) [reply shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js:55]: Card created! https://trello.com/c/7aHigAXh

File: web-client/src/presenter/computeds/partiesInformationHelper.js (hunk 9 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/partiesInformationHelper.js]: Updated!

File: shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js [hunk 10]
```diff
@@ -317,4 +301,86 @@ describe('fileExternalDocumentForConsolidatedInteractor', () => {
...
+    ).toMatchObject({
+      completedMessage: 'completed',
+    });
+  });
+
+  it('should not set as served when is NOT isAutoServed', async () => {
+    applicationContext.getPersistenceGateway().getUserById.mockReturnValue({
+      name: 'Guy Fieri',
+      role: ROLES.petitioner,
+      section: 'docket',
```
Comment (mmarcotte) [inline shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js]: these two still seem at odds

File: web-client/src/presenter/computeds/partiesInformationHelper.js [hunk 11]
```diff
@@ -21,8 +21,35 @@ export const formatCounsel = ({ counsel, screenMetadata }) => {
...
+  const { USER_ROLES } = applicationContext.getConstants();
+
+  if (!petitionIsServed) return false;
+
+  if (user.role === USER_ROLES.petitioner) {
+    return petitioner.contactId === user.userId;
+  }
+
+  if (user.role === USER_ROLES.privatePractitioner) {
+    return userAssociatedWithCase;
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/partiesInformationHelper.js]: continuing to nitpick, I suppose this should be `!!userAssociatedWithCase` to ensure a boolean is returned.

File: shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js (hunk 5 above)
Comment (mmarcotte) [reply shared/src/business/useCases/caseAssociation/deleteCounselFromCaseInteractor.js:55]: oh thank you

File: web-client/src/presenter/computeds/partiesInformationHelper.js (hunk 11 above)
Comment (matthopson) [reply web-client/src/presenter/computeds/partiesInformationHelper.js]: In this instance, that value appears to be passed in as an argument, so shouldn't we assume it's correctly typed?

Comment (matthopson) [reply web-client/src/presenter/computeds/partiesInformationHelper.js]: And that actually comes from here, where it is being explicitly set to a boolean
https://github.com/ustaxcourt/ef-cms/pull/1568/files/d4d3173175b900f1484d95672d68debf593ef827#diff-07f693d2d51e975acae3d851eedd7e0ca12607218755e522cd377f63bbe28f6fR107

```
const userAssociatedWithCase = !!formattedPrivatePractitioners.find(
  practitioner =>
    user.barNumber === practitioner.barNumber &&
     practitioner.representing.includes(petitioner.contactId),
);
```

File: shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js (hunk 10 above)
Comment (matthopson) [reply shared/src/business/useCases/externalDocument/fileExternalDocumentForConsolidatedInteractor.test.js]: Addressed!

File: web-client/src/presenter/computeds/partiesInformationHelper.js (hunk 11 above)
Comment (matthopson) [reply web-client/src/presenter/computeds/partiesInformationHelper.js]: Updated.

**Review by mmarcotte — APPROVED** [review]
yay thanks

Awesome, thanks @matthopson, @ericsorenson, and the rest of the team!

**Pull request merged**